sql/sqlcheck/sqlcheck.go

```diff
@@ -133,6 +133,56 @@ var Destructive = &driverAware{
 	},
 }
 
+// Renames detects potential column renames.
+var Renames = &driverAware{
+	run: func(ctx context.Context, diags []Diagnostic, p *Pass) error {
+		for _, sc := range p.File.Changes {
+			for _, c := range sc.Changes {
+				c, ok := c.(*schema.ModifyTable)
+				if !ok {
+					continue
+				}
+			Loop:
+				for i := range c.Changes {
+					d, ok := c.Changes[i].(*schema.DropColumn)
+					if !ok {
+						continue
+					}
+					for j := range c.Changes[i+1:] {
```

Comment:
Please, document why we only look forward.
https://github.com/ariga/atlas/blob/master/sql/internal/sqlx/diff.go#L188-L193